GameEngine/Agents/SearchNode.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+
+namespace GameEngine.Agents
+{
+    public class SearchNode
+    {
+        public GameState State { get; protected set; }
+        public SearchNode Parent { get; protected set; }
+        public Play Action { get; protected set; }
+        public int PathCost { get; protected set; }
+        public int StepCost { get { return 1; } }
```

Comment:
I feel like step cost is a property of the action, rather than the search node itself.

Reply:
I don't particularly care where the property lives if it's going to be just a constant 1. Do you anticipate a need for step cost to be more complicated than that❔ We could also just omit both `PathCost` and `StepCost` for now since we're not using them yet. I'll move the `StepCost` to the `Play` if we don't.

Reply:
Done.